src/odemis/gui/cont/tabs/fastem_main_tab.py

```diff
@@ -0,0 +1,201 @@
+        for vaname in getVAs(main_data.sed):
+            if vaname in hwdet_vanames:
+                hwdetvas.add(vaname)
+
+        sem_stream = FastEMSEMStream(
+            "Single Beam",
+            main_data.sed,
+            main_data.sed.data,
+            main_data.ebeam,
+            focuser=main_data.ebeam_focus,
+            hwemtvas=hwemtvas,
+            hwdetvas=hwdetvas,
+            emtvas=emtvas,
+        )
+        tab_data.streams.value.append(sem_stream)  # it should also be saved
+        tab_data.semStream = sem_stream
+
+        # Handle the parent panel's size event and set the size of children panel accordingly
+        panel.pnl_user_settings.Bind(wx.EVT_SIZE, self.on_pnl_user_settings_size)
+        panel.pnl_tabs.Bind(wx.EVT_SIZE, self.on_pnl_tabs_size)
+
+        user_settings_panel = main_xrc.xrcpnl_fastem_user_settings(panel.pnl_user_settings)
+        self.user_settings_panel = FastEMUserSettingsPanel(user_settings_panel, main_data)
+
+        overview_panel = main_xrc.xrcpnl_tab_fastem_overview(panel.pnl_tabs)
+        self.overview_tab = FastEMOverviewTab(
+            "fastem_overview",
+            panel.btn_tab_overview,
+            overview_panel,
+            main_frame,
+            main_data,
+            self.vp,
+            self.view_controller,
+            sem_stream,
+        )
+
+        acquisition_panel = main_xrc.xrcpnl_tab_fastem_acqui(panel.pnl_tabs)
+        self.acquisition_tab = FastEMAcquisitionTab(
+            "fastem_acq",
+            panel.btn_tab_acqui,
+            acquisition_panel,
+            main_frame,
+            main_data,
+            self.vp,
+            self.view_controller,
+            sem_stream,
+        )
+
+        self._tab = model.VAEnumerated(None, choices={None: ""})
+        self.tab_controller = TabController(
+            [self.overview_tab, self.acquisition_tab],
+            self._tab,
+            main_frame,
+            main_data,
+            self.overview_tab,
+        )
+
+        panel.btn_pnl_user_settings.Bind(wx.EVT_BUTTON, self._toggle_user_settings_panel)
+
+    def on_pnl_user_settings_size(self, evt):
+        """Handle the wx.EVT_SIZE event for pnl_user_settings"""
+        self.user_settings_panel.panel.SetSize(
+            (-1, self.user_settings_panel.panel.Parent.GetSize().y)
+        )
+        self.user_settings_panel.panel.Layout()
+        self.user_settings_panel.panel.Refresh()
+        evt.Skip()
+
+    def on_pnl_tabs_size(self, evt):
+        """Handle the wx.EVT_SIZE event for pnl_tabs"""
+        self.overview_tab.panel.SetSize((-1, self.overview_tab.panel.Parent.GetSize().y))
+        self.acquisition_tab.panel.SetSize((-1, self.acquisition_tab.panel.Parent.GetSize().y))
+        self.overview_tab.panel.Layout()
+        self.acquisition_tab.panel.Layout()
+        self.overview_tab.panel.Refresh()
+        self.acquisition_tab.panel.Refresh()
+        evt.Skip()
+
+    @call_in_wx_main
```

Comment:
It's a callback from wxPython. You must *not* protect it, as otherwise it can cause issues. It is always called in the GUI thread, and should be immediately run (not postponed, as this decorator does).